### ISSUE TITLE
Allow user to filter lineages in tree by most recent collection date

### ISSUE THREAD

**Issue by ArtPoon** [PoonLab/covizu#332]
A substantial number of lineages are no longer observed - it would save a lot of space to filter them out, but to enable the user to view these lineages for historical context or research applications.
We should also restore the vertical scroll bar for tall trees.  I'd prefer this to be on the left side.

**Comment by ArtPoon** [comment]
Since we have to manipulate the tree structure in order to redraw it according to which tips are selected (by collection date), we should make a second copy of the tree so that we are not modifying the version loaded  from the server.  Note that the `fortify` function converts the Newick tree into a "data frame"-like object that we can filter and then renumber tips based on what's left.  (We'd remove internal nodes that have no descendants in the filtered subset.)

**Comment by ArtPoon** [comment]
There should already be a function for assigning vertical (y) coordinates based on how we traverse the tree from tips to the root.

**Comment by ArtPoon** [comment]
Regarding the interface for selecting a date cutoff, I think it would be the most effective to have a horizontal slider aligned with the date axis of the following panel:
<img width="259" alt="Screen Shot 2022-03-08 at 3 49 08 PM" src="https://user-images.githubusercontent.com/1109328/157322543-ef205f0c-7e2b-4e26-9568-7fc48cfe0261.png">

and when the user grabs the handle of the slider, an animated vertical line can be drawn over the tree as feedback on where the cutoff would take effect.  Releasing the handle should redraw the tree excluding tips where the most recent sample date is earlier than the cutoff.  We should not rescale the horizontal axis (time), even if excluding "old lineages" changes the location of the root (it shouldn't).
Then if user clicks on a "reset" button the panel should restore the original tree.

**Comment by bonnielu** [comment]
I’ve written a function that will filter/collapse the `root` that gets generated at `readTree()`. This filtered root then gets passed along through `readTimeTree()` where the new coordinates get calculated and a new`df` gets generated. This, however, modifies the original root. 

I thought that I would be able to run a deep copy on it, but that doesn't work since root contains circular references. So we’d have to write our own function to copy that object and then call it every time the user wants to filter the tree. Alternatively, we could reread the newick string each time, but both of those options don’t sound great. 

I think that I'm going to try filtering/collapsing the tree from `df` and write a function to reassign the y-coordinates at that level. I was initially worried about this creating redundancy, but this is probably the lesser evil. 

If anyone has any other suggestions on how best to approach this, that’d be much appreciated!

**Comment by ArtPoon** [comment]
If the user wants to run a search for a specific lineage or a set of samples, we should reset the tree filter to the default (entire tree).  We should also deactivate the slider widget associated with this filter while a search is active (i.e., until the user hits the Clear button).

**Comment by ArtPoon** [comment]
@bonnielu has there been any progress on this issue?  thanks!

**Comment by bonnielu** [comment]
Hi! Sorry for the delay on this issue, I got caught up with the indel modeling. As of now, I've refactored the filtering function so that it builds a subset from the `df` rather than pruning from the `root` variable. This makes it compatible with the recent changes and also results in a faster processing time. There are still a few kinks that I need to work out, but this feature should be done before the start of next week!
